modules/components.js

```diff
@@ -0,0 +1,83 @@
+export function createPostElement(post, options = {}) {
+    const container = document.createElement("article")
+    container.className = "post"
+    container.setAttribute("data-id", post.id)
+    container.onclick = () => options.onClick?.(post)
+
+    const header = document.createElement("section")
+    header.className = "post__header"
+    container.append(header)
+
+    const title = document.createElement("h1")
+    title.className = "post__title"
+    title.innerText = post.title
+    header.append(title)
+
+    const actions = document.createElement("div")
+    actions.className = "post__actions"
+    header.append(actions)
+
+    if (options.canEdit) {
```

Comment:
The delete and edit buttons appear for everyone even if they don't own the post.